mixins/localize/localize-mixin.js

```diff
@@ -6,100 +6,74 @@ import { html } from 'lit';
+	return new RegExp(`<(?!${allowedAfterTriangleBracket})`);
+};
 
-	connectedCallback() {
-		super.connectedCallback();
-		this.constructor.documentLocaleSettings.addChangeListener(this.__languageChangeCallback);
-		this.__languageChangeCallback();
-	}
+const disallowedTagsRegex = getDisallowedTagsRegex(allowedTags);
+const noAllowedTagsRegex = getDisallowedTagsRegex([]);
 
-	disconnectedCallback() {
-		super.disconnectedCallback();
-		this.constructor.documentLocaleSettings.removeChangeListener(this.__languageChangeCallback);
-		this.__updatedProperties.clear();
-	}
+const getLocalizeClass = (superclass = class {}) => class LocalizeClass extends superclass {
 
-	async getUpdateComplete() {
-		await super.getUpdateComplete();
-		const hasResources = this._hasResources();
-		const resourcesLoaded = this.__resources !== undefined;
-		if (!hasResources || resourcesLoaded) {
-			return;
+	static documentLocaleSettings = getDocumentLocaleSettings();
+	#resourcesPromise;
+	pristine = true;
+	#resolveResourcesLoaded;
+
+	#localeChangeCallback;
+	async #defaultLocaleChangeCallback() {
+		if (!this._hasResources()) return;
+
+		const resourcesPromise = this.constructor._getAllLocalizeResources(this.config);
+		this.#resourcesPromise = resourcesPromise;
+		const localizeResources = (await resourcesPromise).flat(Infinity);
+		if (this.#resourcesPromise !== resourcesPromise) return;
+
+		const allResources = {};
+		const resolvedLocales = new Set();
+		for (const { language, resources } of localizeResources) {
+			for (const [key, value] of Object.entries(resources)) {
+				allResources[key] = { language, value };
+				resolvedLocales.add(language);
+			}
 		}
-		await this.__resourcesLoadedPromise;
-	}
-
-	shouldUpdate(changedProperties) {
-
-		const hasResources = this._hasResources();
-		if (!hasResources) {
-			return super.shouldUpdate(changedProperties);
+		this.localize.resources = allResources;
+		this.localize.resolvedLocale = resolvedLocales;
+		if (resolvedLocales.size > 1) {
+			console.warn(`Resolved multiple locales: ${[...resolvedLocales].join(', ')}`);
 		}
 
-		const ready = this.__resources !== undefined;
-		if (!ready) {
-			changedProperties.forEach((oldValue, propName) => {
-				this.__updatedProperties.set(propName, oldValue);
-			});
-			return false;
+		this.#onResourcesChange();
+
+		if (this.pristine) {
+			this.pristine = false;
+			this.#resolveResourcesLoaded();
 		}
+	}
 
-		this.__updatedProperties.forEach((oldValue, propName) => {
-			if (!changedProperties.has(propName)) {
-				changedProperties.set(propName, oldValue);
-			}
-		});
-		this.__updatedProperties.clear();
+	#onResourcesChange() {
+		this.dispatchEvent?.(new CustomEvent('d2l-localize-resources-change'));
+		this.config?.onResourcesChange?.();
+		this.onLocalizeResourcesChange?.();
+	}
 
-		return super.shouldUpdate(changedProperties);
+	connect(cb = () => this.#defaultLocaleChangeCallback()) {
```

Comment:
Honestly I don't remember why I made the callback configurable. Can probably always be what is now the "default".